SourceWriters/CommonWriter.cs

```diff
@@ -0,0 +1,930 @@
+                    PropertyPart += "        }\r\n";
+                    PropertyPart += "\r\n";
+                }
+            }
+
+            // if we have a random number generator control, include the following lines...
+            if (UserControls.Any(u => u.ElementType == ElementType.ReseedButton))
+            {
+                PropertyPart += "        [ThreadStatic]\r\n";
+                PropertyPart += "        private static Random RandomNumber;\r\n";
+                PropertyPart += "\r\n";
+                PropertyPart += "        private int randomSeed;\r\n";
+                PropertyPart += "        private int instanceSeed;\r\n";
+                PropertyPart += "\r\n";
+            }
+
+            // generate OnCreatePropertyCollection()
+            PropertyPart += "\r\n";
+
+            if (isFileType)
+            {
+                PropertyPart += "        public override PropertyCollection OnCreateSavePropertyCollection()\r\n";
+            }
+            else
+            {
+                PropertyPart += "        protected override PropertyCollection OnCreatePropertyCollection()\r\n";
+            }
+
+            PropertyPart += "        {\r\n";
+
+            // Check to see if we're including a color wheel without an alpha slider
+            if (UserControls.Any(u => u.ElementType == ElementType.ColorWheel && !u.ColorWheelOptions.HasFlag(ColorWheelOptions.Alpha)))
+            {
+                PropertyPart += "            ColorBgra PrimaryColor = EnvironmentParameters.PrimaryColor.NewAlpha(byte.MaxValue);\r\n";
+                PropertyPart += "            ColorBgra SecondaryColor = EnvironmentParameters.SecondaryColor.NewAlpha(byte.MaxValue);\r\n";
+                PropertyPart += "\r\n";
+            }
+
+            // Check to see if we're including a user selectable blending mode
+            if (UserControls.Any(u => u.ElementType == ElementType.BinaryPixelOp))
+            {
+                PropertyPart += "            // setup for a user selected blend mode\r\n";
+                PropertyPart += "            IEnumLocalizerFactory factory = Services.GetService<IEnumLocalizerFactory>();\r\n";
+                PropertyPart += "            IEnumLocalizer blendModeLocalizer = factory.Create(typeof(LayerBlendMode));\r\n";
+                PropertyPart += "            IList<ILocalizedEnumValue> blendModes = blendModeLocalizer.GetLocalizedEnumValues();\r\n";
+                PropertyPart += "            object[] blendModesArray = blendModes.Select(lev => lev.EnumValue).ToArrayEx();\r\n";
+                PropertyPart += "            int defaultBlendModeIndex = blendModesArray.IndexOf(LayerBlendMode.Normal);\r\n";
+                PropertyPart += "\r\n";
+            }
+
+            if (UserControls.Any(u => u.ElementType == ElementType.FontFamily))
+            {
+                PropertyPart += "            FontFamily[] installedFontFamilies = new InstalledFontCollection().Families;\r\n";
```

Comment:
This will of course be different for DirectWrite but it's still possible. 

`IFontCollection IDirectWriteFactory.GetSystemFontCollection()` (DW's built-in system font list thingy)
`ISystemFonts DirectWriteFactoryExtensions.GetSystemFonts()` (something that PDN provides, mostly for UI)
`IGdiFontMap DirectWriteFactoryExtensions.GetGdiFontMap(bool hideInactiveFonts)` (another thing PDN provides)